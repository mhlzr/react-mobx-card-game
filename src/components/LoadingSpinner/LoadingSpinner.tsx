import React, { ReactElement, FunctionComponent, ReactNode } from 'react';
import styled, { keyframes } from 'styled-components';


const animation = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

const Wrapper = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
`;

const Circle = styled.div`
    animation: ${animation} 0.8s infinite ease-in-out;
    border-radius: 50%;
    border: 0.1em solid transparent;
    border-right-color: currentColor;
    color: inherit;
    font-size: 1.5em;
    height: 0.8em;
    margin-left: -0.5em;
    width: 0.8em;
    will-change: transform;
`;

interface LoadingSpinnerProps {
    className?: string;
    children?: ReactNode;
}

export const LoadingSpinner: FunctionComponent<LoadingSpinnerProps> = ({ className, children }: LoadingSpinnerProps): ReactElement => {
    return (
        <Wrapper className={className}>
            <Circle />
            {children}
        </Wrapper>
    )
}



