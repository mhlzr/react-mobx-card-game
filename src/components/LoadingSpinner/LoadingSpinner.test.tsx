import React from 'react';
import { shallow } from 'enzyme'

import { LoadingSpinner } from './LoadingSpinner';

describe('LoadingSpinner', () => {
    it('renders without a label if none is passed', () => {
        const wrapper = shallow(<LoadingSpinner></LoadingSpinner>);
        expect(wrapper.text()).toHaveLength(0);
    });

    it('renders with a label if one is passed', () => {
        const wrapper = shallow(<LoadingSpinner>Test</LoadingSpinner>);
        expect(wrapper.text()).toEqual('Test');
    });

    it('renders with child-components', () => {
        const wrapper = shallow(<LoadingSpinner><span>Child</span></LoadingSpinner>);
        expect(wrapper.find('span')).toHaveLength(1);
        expect(wrapper.find('span').text()).toEqual('Child');
    });

    it('has a snapshot', () => {
        const wrapper = shallow(<LoadingSpinner></LoadingSpinner>);
        expect(wrapper).toMatchSnapshot();
    });
})
