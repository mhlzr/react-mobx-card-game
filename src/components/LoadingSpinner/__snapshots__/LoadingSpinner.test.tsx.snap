// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner has a snapshot 1`] = `
<styled.div>
  <styled.div />
</styled.div>
`;
